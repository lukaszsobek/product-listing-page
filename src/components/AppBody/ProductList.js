import React from "react";

const ProductList = props => {
    const { productList } = props;

    const listItems = productList.map((productData, key) => {
        return <ProductListItem key={key} data={productData} />
    });

    return (
        <section className="product-list">
            { listItems }
        </section>
    );
};

const ProductListItem = props => {
    const { name, category, colors, price } = props.data;

    return (
        <div className="product-list__item">
            <div className="product-list__item__name">{ name }</div>
            <div className="product-list__item__category">Category: { category }</div>
            <div className="product-list__item__colors">
                Colors: <ColorList data={colors} />
            </div>
            <div className="product-list__item__price">Price: &euro; { price }</div>
        </div>
    );
}

const ColorList = props => {
    return props.data.map((color, key) => {
        return (
            <span key={key} className="product-list__item__colors__name" >
                { color }
            </span>
        );
    });
};

export { ColorList, ProductListItem, ProductList };
export default ProductList;