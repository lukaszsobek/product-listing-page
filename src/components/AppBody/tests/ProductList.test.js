import { ColorList, ProductListItem, ProductList } from "../ProductList";


describe("ProductList", () => {

    it("renders ProductList",() => {
        const props = {
            productList: [1,2]
        }
        expect(ProductList(props)).toMatchSnapshot();
    });

    it("renders ProductListItem",() => {

        const props = {
            data: {
                name: "name",
                category: "category",
                colors: "colors",
                price: "price" }
        }
        expect(ProductListItem(props)).toMatchSnapshot();
    });

    it("renders ColorList",() => {

        const props = {
            data: [1,2]
        }
        expect(ColorList(props)).toMatchSnapshot();
    });
});