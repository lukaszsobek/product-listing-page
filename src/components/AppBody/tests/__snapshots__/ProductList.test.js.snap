// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList renders ColorList 1`] = `
Array [
  <span
    className="product-list__item__colors__name"
>
    1
</span>,
  <span
    className="product-list__item__colors__name"
>
    2
</span>,
]
`;

exports[`ProductList renders ProductList 1`] = `
<section
  className="product-list"
>
  <ProductListItem
    data={1}
  />
  <ProductListItem
    data={2}
  />
</section>
`;

exports[`ProductList renders ProductListItem 1`] = `
<div
  className="product-list__item"
>
  <div
    className="product-list__item__name"
  >
    name
  </div>
  <div
    className="product-list__item__category"
  >
    Category: 
    category
  </div>
  <div
    className="product-list__item__colors"
  >
    Colors: 
    <ColorList
      data="colors"
    />
  </div>
  <div
    className="product-list__item__price"
  >
    Price: â‚¬ 
    price
  </div>
</div>
`;
